packages/front-end/components/Settings/EditLicenseModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { FC, useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { FaExclamationTriangle } from "react-icons/fa";
+import { useAuth } from "@/services/auth";
+import { isCloud } from "@/services/env";
+import { useUser } from "@/services/UserContext";
+import Field from "../Forms/Field";
+import Modal from "../Modal";
+import styles from "./EditLicenseForm.module.scss";
+
+const EditLicenseModal: FC<{
+  close: () => void;
+  mutate: () => Promise<unknown>;
+}> = ({ close, mutate }) => {
+  const { license, updateUser } = useUser();
+  const { apiCall } = useAuth();
+  const [closeCta, setCloseCta] = useState("Cancel");
+  const [successMessage, setSuccessMessage] = useState("");
+  const [editField, setEditField] = useState(!license);
+
+  const form = useForm({
+    defaultValues: {
+      licenseKey: "",
+    },
+  });
+
+  useEffect(() => {
+    if (isCloud()) {
+      close();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCloud]);
+
+  if (isCloud()) {
+    return null;
+  }
```

Comment:
Minor nitpick but I would probably leave the above `isCloud` checks out ... The parent component should be responsible of deciding whether to display the modal ... This can help add some 'safety' but it's marginal and the code could be cleaner as a result

Reply:
Point taken, but unless a strong objection I'd like to leave it in. It's an additional check that prevents somebody from implementing this modal somewhere else and having it do something destructive to the org record. (We do also block on the backend, but this way it's at least more obvious that we shouldn't be using it)